Add nimble-select Angular guidance for compareWith function (#2699)
ni/nimble: packages/storybook/src/nimble/select/select.mdx

<Description of={listOptionGroupStories} />
<Controls of={listOptionGroupStories.listOptionGroup} />

> ### Angular

Reviewer: Looks like the example app follows this pattern. Should we update it to obey the new guidance? Or update the guidance to explain whatever subtlety makes it so the bug doesn't occur in that case?

https://github.com/ni/nimble/blob/3b0f6f6a149e769fb040b5b1331154d9709327fb/packages/angular-workspace/example-client-app/src/app/customapp/customapp.component.html#L199

> ### Angular
> 
> In Angular, an <Tag name={listOptionTag} /> can be created with an `ngValue`. In
> that scenario, especially when using a `filter-mode` of `manual`, a
> [`compareWith` implementation](https://angular.dev/api/forms/SelectControlValueAccessor#customizing-option-selection)
> should be provided on the Select control. Without this, the Select may display
> unexpected behavior (like reverting to the placeholder value when updating the
> list of filtered options).

Reviewer: Based on [these docs](https://angular.dev/api/forms/NgSelectOption#ngValue) the difference between `value` and `ngValue` is that the latter should be used when the value is an object. I think it'd be helpful to add that context here, either inline or with a link. Maybe something like this?

I'd also suggest moving this inside the `### List Option` section.

```suggestion
### Angular

In Angular, a <Tag name={listOptionTag} /> [can be created](https://angular.dev/api/forms/NgSelectOption) with a string `value` or an object `ngValue`. When using `ngValue`, provide a
[`compareWith` implementation](https://angular.dev/api/forms/SelectControlValueAccessor#customizing-option-selection) on the Select control. Without this, the Select may display
unexpected behavior (like reverting to the placeholder value when updating the
list of filtered options).
```
